src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php

```diff
@@ -260,8 +260,16 @@ private static function resolveValidationGroups($groups, FormInterface $form)
 
     private static function getConstraintsInGroups($constraints, $group)
     {
-        return array_filter($constraints, static function (Constraint $constraint) use ($group) {
-            return \in_array($group, $constraint->groups, true);
+        $groups = (array) $group;
```

Comment:
This is coming from \Symfony\Component\Validator\Constraints\GroupSequence::$groups and according to it's type, $group itself could also be a GroupSequence again. So there is still something not right.

Reply:
That's true, yet nobody reported this as a bug ever, so that the urgency on this is zero on my side :)

Reply:
Using a nested array seems to be the same as using a nested GroupSequence. If the nested group sequence is not supported, we probably should just remove the type from GroupSequence::$groups. It's also not documented in https://symfony.com/doc/current/validation/sequence_provider.html